r/R/filesystem.R

```diff
@@ -108,7 +108,10 @@ Selector <- R6Class("Selector",
 )
 
 Selector$create <- function(base_dir, allow_non_existent = FALSE, recursive = FALSE) {
-  shared_ptr(Selector, fs___Selector__create(base_dir, allow_non_existent, recursive))
+  shared_ptr(
+    Selector,
+    fs___Selector__create(fs_relative_path(base_dir), allow_non_existent, recursive)
```

Comment:
Are you sure? Will it work correctly if `base_dir` is e.g. a path in an S3 filesystem, or will it produce bogus results?

Reply:
Same question for all such cases below.